pkg/controller/node/cidr_set.go

```diff
@@ -38,10 +38,16 @@ type cidrSet struct {
 func newCIDRSet(clusterCIDR *net.IPNet, subNetMaskSize int) *cidrSet {
 	clusterMask := clusterCIDR.Mask
 	clusterMaskSize, _ := clusterMask.Size()
-	maxCIDRs := 1 << uint32(subNetMaskSize-clusterMaskSize)
+
+	var maxCIDRs int
```

Comment:
@smarterclayton I updated the review, for the comments on the allocation logic and the bitmap I have a check to limit the bitmap size. Let me know if this addresses your concern. Also let me know if this should be applied to v4 as well as v6.